### DISCUSSION TITLE
Example using vendor-dependencies

### DISCUSSION THREAD

**Discussion by helio-frota** [guacsec/trustify#513] [Show and tell]
Hi, I created this repository to show how we can use rust vendor-dependencies and also validate that we are really using them.

The example uses `cargo vendor` and `cargo cache` commands

https://github.com/helio-frota/vendedor

( Apparently not possible to vendor specific dependencies https://github.com/rust-lang/cargo/issues/9234 )

**Comment by bobmcwhirter** [comment]
Thanks!

Wouldja mind also testing how vendoring a `git` dependency works? I know we shouldn't be releasing with them, but who knows what the future may ever bring, so understanding how vendoring works with things from not-crates.io would be excellent.

**Reply by helio-frota** [reply]
sure thing :+1: 
you probably refers to "private registries"
I'll take a look on that

**Reply by bobmcwhirter** [reply]
more when Cargo.toml specifies something like

```
foo = { git = "https://....", commit = "123123213123" }
```

**Reply by helio-frota** [reply]
sorry the long delay 

I've updated the example to use 2 dependencies: `zero` and `colored_json` to make things more clear with less sub-dependencies

* We can rename dependencies and use vendor 
  * https://github.com/helio-frota/vendedor/blob/main/Cargo.toml#L7-L8
  * https://github.com/helio-frota/vendedor/blob/main/.cargo/config.toml#L4-L7
* When using `foo { git } ` in `Cargo.toml` we also need to add the git information in `config.toml`
  * https://github.com/helio-frota/vendedor/blob/main/Cargo.toml#L10
  * https://github.com/helio-frota/vendedor/blob/main/.cargo/config.toml#L9-L10
* Both zero and colored_json end up inside `vendor` directory

**Reply by helio-frota** [reply]
> we also need to add the git information in config.toml

```shell
➜  vendedor git:(main) ✗ cargo build
error: no matching package found
searched package name: `serde`
perhaps you meant:      zero
location searched: registry `crates-io`
required by package `colored_json v5.0.0 (https://github.com/ctron/colored_json#962ed177)`
    ... which satisfies git dependency `colored_json` (locked to 5.0.0) of package `vendedor v0.1.0 (/home/heliofrota/Desktop/vendedor)`
```
